proc_netstat.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2022 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procfs
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/mitchellh/mapstructure"
```

Comment:
Hmm, I'm not sure we want to add this dependency. But maybe it would be useful for other stuff.

CC @discordianfish @roidelapluie

Reply:
Also, we split stdlib and external deps like this:

```
import (
	"fmt"
	"os"

	"github.com/mitchellh/mapstructure"
)
```

Reply:
@SuperQ I resolved the rest of the comments and amend the last commit. 
How i should proceed with the `mapstructure` dependency?

Reply:
Hrm yeah not sure either. To get this merged asap, I'd remove the dependency assign manually to the struct for now.

But I see something like that being useful for similar cases... If we want to do that, we should right away use this for these cases to establish that pattern for further contributions.

Reply:
@discordianfish Which are those cases?

Reply:
actually indeed there is a lot verbosity on the parsing of map to structure e.g. [here](https://github.com/prometheus/procfs/blob/master/blockdevice/stats.go#L346). Maybe we can merge this PR and i can start asap working on replacing those verbose parsing cases with `mapstructure`?

Reply:
I think this deserve a longer discussion. So if you want this to get merged soon, remove mapstructure and assign these manually.